api/controllers/webhook.js

```diff
@@ -84,50 +100,45 @@ const createBuildForWebhookRequest = async (request) => {
   if (queuedBuild) {
     return queuedBuild.update({
       commitSha,
-      user: user.id,
+      user: user ? user.id : null,
+      username,
     });
   }
 
   return Build.create({
     branch,
     commitSha,
     site: site.id,
-    user: user.id,
+    user: user ? user.id : null,
+    username,
   })
     .then(build => build.enqueue());
 };
 
 module.exports = {
-  github(req, res) {
-    signWebhookRequest(req)
-      .then(() => {
-        if (req.body.commits && req.body.commits.length > 0) {
-          return createBuildForWebhookRequest(req);
-        }
-
-        return null;
-      })
-      .then((build) => {
-        if (!build) {
-          res.ok('No new commits found. No build scheduled.');
-          return null;
-        }
-
-        return GithubBuildStatusReporter.reportBuildStatus(build)
-          .then(() => buildSerializer.serialize(build));
-      })
-      .then((buildJSON) => {
-        if (buildJSON) {
-          res.json(buildJSON);
-        }
-      })
-      .catch((err) => {
-        if (err.message) {
-          res.badRequest(err);
-        } else {
-          logger.error(err);
-          res.badRequest();
-        }
-      });
-  },
+  github: (req, res) => signWebhookRequest(req)
+    .then(() => {
+      if (req.body.commits && req.body.commits.length > 0) {
+        return createBuildForWebhookRequest(req);
+      }
+      return Promise.resolve();
```

Comment:
not necessary to return a "Promise", this is the same as `return undefined` but with more work.

Reply:
linting rules require something to be returned